src/vm/hints/hint_utils.rs

```diff
@@ -277,6 +279,135 @@ pub fn assert_le_felt(
+                        maybe_rel_a.clone(),
+                        maybe_rel_b.clone(),
+                    ));
+                };
+                Ok(())
+            }
+            (MaybeRelocatable::RelocatableValue(a), MaybeRelocatable::RelocatableValue(b)) => {
+                if a.segment_index != b.segment_index {
+                    return Err(VirtualMachineError::DiffIndexComp(a.clone(), b.clone()));
+                };
+                if a.offset == b.offset {
+                    return Err(VirtualMachineError::AssertNotEqualFail(
+                        maybe_rel_a.clone(),
+                        maybe_rel_b.clone(),
+                    ));
+                };
+                Ok(())
+            }
+            _ => Err(VirtualMachineError::DiffTypeComparison(
+                maybe_rel_a.clone(),
+                maybe_rel_b.clone(),
+            )),
+        },
+        _ => Err(VirtualMachineError::FailedToGetIds),
+    }
+}
+
+//Implements hint:
+// %{
+//     from starkware.cairo.common.math_utils import assert_integer
+//     assert_integer(ids.a)
+//     assert 0 <= ids.a % PRIME < range_check_builtin.bound, f'a = {ids.a} is out of range.'
+// %}
+pub fn assert_nn(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
+) -> Result<(), VirtualMachineError> {
+    //Check that ids contains the reference id for 'a' variable used by the hint
+    let a_ref = if let Some(a_ref) = ids.get(&String::from("a")) {
+        a_ref
+    } else {
+        return Err(VirtualMachineError::IncorrectIds(
+            vec![String::from("a")],
+            ids.into_keys().collect(),
+        ));
+    };
+    //Check that 'a' reference id corresponds to a value in the reference manager
+    let a_addr =
+        if let Some(a_addr) = get_address_from_reference(a_ref, &vm.references, &vm.run_context) {
+            a_addr
+        } else {
+            return Err(VirtualMachineError::FailedToGetIds);
+        };
+
+    //Check that the 'a' id is in memory
+    let maybe_rel_a = if let Ok(Some(maybe_rel_a)) = vm.memory.get(&a_addr) {
+        maybe_rel_a
+    } else {
+        return Err(VirtualMachineError::FailedToGetIds);
+    };
+
+    //assert_integer(ids.a)
+    let a = if let &MaybeRelocatable::Int(ref a) = maybe_rel_a {
+        a
+    } else {
+        return Err(VirtualMachineError::ExpectedInteger(a_addr.clone()));
+    };
+
+    for (name, builtin) in &vm.builtin_runners {
+        //Check that range_check_builtin is present
+        if name == &String::from("range_check") {
+            let range_check_builtin = if let Some(range_check_builtin) =
+                builtin.as_any().downcast_ref::<RangeCheckBuiltinRunner>()
+            {
+                range_check_builtin
+            } else {
+                return Err(VirtualMachineError::NoRangeCheckBuiltin);
+            };
+
+            // assert 0 <= ids.a % PRIME < range_check_builtin.bound
+            if a.mod_floor(&vm.prime).is_positive()
```

Comment:
`.mod_floor` is always positive if `vm.prime` is positive.

Reply:
Fixed!